File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x:x[0])
        array=[intervals[0]]
        for i in range(len(intervals)):
            if array[-1][-1]>=intervals[i][0]:
                if array[-1][-1]<intervals[i][-1]:
                    array[-1][-1]=intervals[i][-1]
            else:
                array.append(intervals[i])
        return array
        # res=[intervals[0]]
        # j=0
        # for i in range(1,len(intervals)):
        #     if res[j][-1]>=intervals[i][0]:
        #         if res[j][-1]<intervals[i][-1]:
        #             res[j][-1]=intervals[i][-1]
        #         else:
        #             pass
        #     else:
        #         res.append(intervals[i])
        #         j+=1
        # return res

        
        

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int m=intervals.length;
        int n=intervals[0].length;
        int[][] result=new int[m][2];
        int j=0;
        Arrays.sort(intervals,(a,b)->Integer.compare(a[0],b[0]));
        result[0]=intervals[0];
        for(int i=1;i<m;i++){
            if(result[j][1]>=intervals[i][0]){
                result[j][1]=Math.max(result[j][1],intervals[i][1]);
            }else{
                j++;
                result[j]=intervals[i];
            }
        }
        return Arrays.copyOf(result,j+1);
    }
}